[Patterns]: Add new `WooCommerce 2-by-2 Product Feature Grid` pattern

albarin opened the pull request:
This PR implements the `WooCommerce 2-by-2 Product Feature Grid` pattern.

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/9052

### Screenshots

| Design | Pattern |
| ------ | ----- |
| ![Rigid grid](https://user-images.githubusercontent.com/186112/232024052-e81ee9c3-74fb-4b4a-927e-3b6cade16157.png) | ![CleanShot 2023-04-17 at 13 57 46@2x](https://user-images.githubusercontent.com/186112/232477593-ba10ba2d-a435-4c1b-af86-bd741bdf6d5f.png) |

### Testing
#### User-Facing Testing
1. Create a new page or post
2. Make sure the `WooCommerce 2-by-2 Product Feature Grid` pattern appears under the WooCommerce category dropdown.
3. Insert in and make sure it shows as expected on the design.

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

> Add the new `WooCommerce 2-by-2 Product Feature Grid` pattern.

albarin committed:
Add new `Product features: rigid grid` pattern

shaunandrews commented on patterns/product-feature-rigid-grid.php:
```diff
@@ -0,0 +1,96 @@
+<div class="wp-block-group alignwide">
+	<!-- wp:heading {"textAlign":"left","level":4,"align":"wide"} -->
+	<h4 class="wp-block-heading alignwide has-text-align-left">
+		<strong>What makes this product unique?</strong>
```
Why the strong tag here?

shaunandrews commented:
Mostly works as expected on my test site. I do have a few questions:

- Should we (could we?) use the Add to Cart Button block in our patterns?
- Should we use real words for the example copy? Other patterns seem to avoid lorem ipsum.
- Speaking of other patterns, they often include imagery in both the pattern preview and when the pattern is added to the canvas. Can we include imagery in either the pattern preview or when the pattern is added, and, can we control these independently? I like the placeholders when the pattern is added, but they make the pattern preview appear lackluster.

A small bug:

<img width="71" alt="CleanShot 2023-04-17 at 10 34 46@2x" src="https://user-images.githubusercontent.com/191598/232520770-97f26939-d6c4-44b2-918c-aeb0cc804f7e.png">

The first Heading has an extraneous space at the beginning. Maybe has to do with the newlines and indenting in the pattern source?

shaunandrews commented:
<img width="1338" alt="CleanShot 2023-04-17 at 11 29 56@2x" src="https://user-images.githubusercontent.com/191598/232535887-2a377934-9b06-4b5d-b430-45566fd21c0f.png">

I updated the Figma to add some more realistic copy across the entire pattern.

<img width="681" alt="CleanShot 2023-04-17 at 11 29 37@2x" src="https://user-images.githubusercontent.com/191598/232535852-3dd18942-a17f-4621-81c5-4c8981110a2e.png">

Can we also rename this? "Rigid" is a weird word to use in this context, and the whole name is rather long. Maybe something like "2-by-2 Product Feature Grid" would be a little nicer?

albarin committed:
Update title, images and copy

albarin commented:
> I updated the Figma to add some more realistic copy across the entire pattern.

I've updated the pattern too.

> Can we also rename this? "Rigid" is a weird word to use in this context, and the whole name is rather long. Maybe something like "2-by-2 Product Feature Grid" would be a little nicer?

Yes, I was going to ask about it, I was using the names from Figma temporarily, I've updated it with your suggestion.

I've also added some example images that come with WC Core.

What do you think @shaunandrews?

albarin committed:
Stop hardcoding urls
